src/gnc/attitude_estimator.cpp

```diff
@@ -438,10 +439,35 @@ void attitude_estimator_reset(AttitudeEstimatorState &state,
     double t, lin::Vector3d const &r_ecef, lin::Vector3f const &b_body,
     lin::Vector3f const &s_body) {
   GNC_ASSERT_NORMALIZED(s_body);
-
-  // TODO : Implement this
-  // https://github.com/pathfinder-for-autonomous-navigation/psim/issues/187
+  
   lin::Vector4f q_body_eci = lin::nans<float, 4, 1>();
+  
+  lin::Vector3f b_eci{};
+  lin::Vector4f q{};
+  env::magnetic_field(t, lin::Vector3f(r_ecef), b_eci);  // b_eci = b_ecef
+  env::earth_attitude(t, q);                             // q = q_ecef_eci
+  utl::quat_conj(q);                                     // q = q_eci_ecef
+  utl::rotate_frame(q, b_eci);                           // b_eci = b_eci  
+  env::magnetic_field(t, lin::Vector3f(r_ecef), b_eci);
+  
+  // Ensure the measured and expected magnetic field are large enough
+  float thresh = constant::b_noise_floor * constant::b_noise_floor;
```

Comment:
We should throw this into something like:
```
{
  float thresh = constant::b_noise_floor * constant::b_noise_floor;
  float fro_b_eci = lin::fro(b_eci);
  if (...) return;
}
```
The compiler may see through it but just in case it doesn't we can save some floating point multiplications.